### PR TITLE
Add `dispose` method and backfill more tests for `createClient`

### PR DESCRIPTION
**Pull request by mthadley**
Adds a test suite to cover at least the basics of `createClient`. There's still more to add like better coverage for the refresh loop itself, which will come later.

### CODE REVIEW

File: src/create-client.ts [hunk 1]
```diff
@@ -328,5 +333,6 @@ An authorization_code was supplied for a login which did not originate at the ap
     getUser,
     getAccessToken,
     signOut,
+    dispose,
```
Comment (mthadley) [inline src/create-client.ts:336]: This is the only public API change. I think it makes sense to have, though if anyone has concerns, I can probably find a way to keep it private for the moment.

File: src/create-client.test.ts [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+    );
+
+  describe("createClient", () => {
+    describe("with no `clientId` provided", () => {
+      it("throws an error", async () => {
+        await expect(createClient("")).rejects.toThrow(/Missing Client ID/);
+      });
+    });
+
+    describe("when the current route matches the redirect URI", () => {
```
Comment (cmatheson) [inline src/create-client.test.ts:47]: i think maybe it's more accurate to say 'when a code param is present` or something like that?

**Review by cmatheson — APPROVED** [review]
tests 🎈🥳 🎉

File: src/create-client.test.ts (hunk 2 above)
Comment (mthadley) [reply src/create-client.test.ts:47]: Very true. Will update those test names to be more specific.

**Pull request merged**